sdk/core/core/src/az_client_secret_credential.c

```diff
@@ -177,9 +180,31 @@ AZ_INLINE AZ_NODISCARD az_result az_token_credential_update(
 
   {
     az_json_value value;
-    AZ_RETURN_IF_FAILED(az_json_get_object_member(body, AZ_STR("access_token"), &value));
+
+    clock_t lifetime_seconds = 0;
+    if (requested_at > 0) {
+      az_span expires_in = { 0 };
+      if (az_succeeded(az_json_get_object_member(body, AZ_STR("expires_in"), &value))
+          && az_succeeded(az_json_value_get_string(&value, &expires_in))) {
+        clock_t number = 0;
+        for (size_t i = 0; i < expires_in.size; ++i) {
+          uint8_t const c = expires_in.begin[i];
+          if (c >= '0' && c <= '9') {
+            number = number * 10 + (c - '0');
+          } else {
+            number = 0;
+            break;
+          }
```

Comment:
is this some sort of manual atof() / atoi() ?    
can we use those from stdlib?
if not, it might be a good idea to make this reusable someplace as AZ_INLINE span_to_clock_t(....)

Reply:
Done.